MediaBrowser.Providers/Trickplay/TrickplayManager.cs

```diff
@@ -0,0 +1,385 @@
+                container,
+                mediaSource,
+                mediaStream,
+                width,
+                TimeSpan.FromMilliseconds(options.Interval),
+                options.EnableHwAcceleration,
+                options.ProcessThreads,
+                options.Qscale,
+                options.ProcessPriority,
+                _encodingHelper,
+                cancellationToken).ConfigureAwait(false);
+
+            if (string.IsNullOrEmpty(imgTempDir) || !Directory.Exists(imgTempDir))
+            {
+                throw new InvalidOperationException("Null or invalid directory from media encoder.");
+            }
+
+            var images = _fileSystem.GetFiles(imgTempDir, new string[] { ".jpg" }, false, false)
+                .Where(img => string.Equals(img.Extension, ".jpg", StringComparison.Ordinal))
+                .OrderBy(i => i.FullName)
+                .ToList();
+
+            // Create tiles
+            var tilesTempDir = Path.Combine(imgTempDir, Guid.NewGuid().ToString("N"));
+            var tilesInfo = CreateTiles(images, width, options, tilesTempDir, outputDir);
+
+            // Save tiles info
+            try
+            {
+                if (tilesInfo is not null)
+                {
+                    SaveTilesInfo(video.Id, tilesInfo);
+                    _logger.LogInformation("Finished creation of trickplay files for {0}", mediaPath);
+                }
+                else
+                {
+                    throw new InvalidOperationException("Null trickplay tiles info from CreateTiles.");
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error while saving trickplay tiles info.");
+
+                // Make sure no files stay in metadata folders on failure
+                // if tiles info wasn't saved.
+                Directory.Delete(outputDir, true);
+            }
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error creating trickplay images.");
+        }
+        finally
+        {
+            _resourcePool.Release();
+
+            if (!string.IsNullOrEmpty(imgTempDir))
+            {
+                Directory.Delete(imgTempDir, true);
+            }
+        }
+    }
+
+    private TrickplayTilesInfo CreateTiles(List<FileSystemMetadata> images, int width, TrickplayOptions options, string workDir, string outputDir)
+    {
+        if (images.Count == 0)
+        {
+            throw new InvalidOperationException("Can't create trickplay from 0 images.");
+        }
+
+        Directory.CreateDirectory(workDir);
+
+        var tilesInfo = new TrickplayTilesInfo
+        {
+            Width = width,
+            Interval = options.Interval,
+            TileWidth = options.TileWidth,
+            TileHeight = options.TileHeight,
+            TileCount = 0,
+            Bandwidth = 0
+        };
+
+        var firstImg = SKBitmap.Decode(images[0].FullName);
```

Comment:
Should use `IImageEncoder` instead of referencing Skia directly

Reply:
Wasn't able to figure out a way to do this cleanly. IImageEncoder does not contain a Decode method--only SkiaEncoder does. With the system of ,NET DI I don't believe I can control which implementation of IImageEncoder the class gets. If I do type-checking that sort of breaks the point of the encoder abstraction, and I can't easily import Jellyfin.Drawing.Skia anyways since that seems to be built separately, so importing would break the whole MediaBrowser interface -> Jellyfin impl design.

Really it looks like there's only a SkiaEncoder and a NullImpl when no encoder is available. I could just add Decode to IImageEncoder.

Reply:
> so I could just add Decode to IImageEncoder. What are your thoughts?

Yeah that's what I was thinking. Skia isn't always available on the platform

Reply:
Looked at it again and realized just adding a Decode to IImageEncoder doesn't solve the issue of abstraction since it would have to return a SkiaSharp bitmap, and TrickplayManager still uses things like SkCanvas. Would it be better to move the trickplay tile image creation to IImageEncoder as ```CreateTrickplayTile``` and implement it in SkiaEncoder similar to CreateSplashscreen and CreateImageCollage?

Reply:
Your suggestion of `CreateTrickplayTile` makes sense to me 👍

Reply:
@nicknsy Just an fyi- the only thing blocking approval from me is this thread